### PR TITLE
chore(cherry-pick):759b92e to 12.1.1

### PR DESCRIPTION
**Pull request by montelaidev**
## **Description**

This pr cherry picks the 759b92e. 

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26802?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by Gudahtt** [comment]
@metamaskbot update-policies

File: app/scripts/migrations/105.ts [hunk 1]
```diff
@@ -97,6 +97,7 @@ function createInternalAccountsForAccountsController(
       metadata: {
         name: identity.name,
         lastSelected: identity.lastSelected ?? undefined,
+        importTime: 0,
```
Comment (Gudahtt) [inline app/scripts/migrations/105.ts:100]: Hmm, why was this change required? I'm guessing this was a type error, but would this impact other accounts in state as well? e.g. do we need a migration to add `importTime` to existing account metadata?

Comment (montelaidev) [reply app/scripts/migrations/105.ts:100]: Yea, its a type error. A default importTime was added on migration 119

Comment (Gudahtt) [reply app/scripts/migrations/105.ts:100]: Ah OK, glad it was added later on at least.

Do we need to add it here as well though? This will add `importTime` for users _ahead_ of migration 119, which doesn't seem necessary.

in general I don't think we should use state types like this in migrations because migrations are meant to deal with state at a particular point in time. But types get updated over time. The `InternalAccount` type used in this migration doesn't reflect the shape of an internal account for a user running this migration.

Maybe we can remove the type from the migration instead, or use `// @ts-expect-error` to ignore the error, since it was caused by post-105 type changes

**Comment by montelaidev** [comment]
@SocketSecurity ignore npm/@metamask/snaps-utils@7.8.1

File: ui/components/multichain/pages/connections/connections.tsx [hunk 2]
```diff
@@ -110,7 +110,10 @@ export const Connections = () => {
   );
   const selectedAccount = useSelector(getSelectedAccount);
   const internalAccounts = useSelector(getInternalAccounts);
-  const mergedAccounts = mergeAccounts(connectedAccounts, internalAccounts);
+  const mergedAccounts = mergeAccounts(
+    connectedAccounts,
+    internalAccounts,
+  ) as AccountType[];
```
Comment (Gudahtt) [inline ui/components/multichain/pages/connections/connections.tsx:116]: This type assertion does not appear to be necessary, but at least it's not overriding any valid type errors. Easiest to leave this as-is for the cherry-pick, but please avoid the use of type assertions in the future. See here for details on why, and about alternatives: https://github.com/MetaMask/contributor-docs/blob/main/docs/typescript.md#type-assertions

File: app/scripts/migrations/105.ts (hunk 1 above)
Comment (Gudahtt) [reply app/scripts/migrations/105.ts:100]: It looks like adding `importTime` pre-emptively will not have any negative impact here because migration 119 will still replace it (it replaces any falsy `importTime` metadata, and 0 is falsy). But going forward it would be better to remove state types from migrations rather than updating them after the fact.

File: ui/components/multichain/pages/connections/connections.tsx (hunk 2 above)
Comment (montelaidev) [reply ui/components/multichain/pages/connections/connections.tsx:116]: I think linter was complaining when the pr was made. This type comes from[ this line ](https://github.com/MetaMask/metamask-extension/blob/0490020bf7307d303e958e8cfbb3471b76e24bc9/ui/components/multichain/pages/connections/connections.tsx#L258)

File: ui/pages/remove-snap-account/snap-account-card.tsx [hunk 3]
```diff
@@ -24,7 +25,7 @@ export const SnapAccountCard = ({
   const account = mergedAccounts.find(
     (internalAccount: { address: string }) =>
       internalAccount.address === address,
-  );
+  ) as MergedInternalAccount;
```
[the remark this answers is not in the document]
Comment (montelaidev) [reply ui/pages/remove-snap-account/snap-account-card.tsx:28]: If we show the removal of a snap account, it would mean the account is defined.

File: ui/components/multichain/pages/connections/connections.tsx (hunk 2 above)
Comment (Gudahtt) [reply ui/components/multichain/pages/connections/connections.tsx:116]: Right, but that line is using the wrong type (it's using `AccountType`, but this is a `MergedAccountType`). If the type is corrected, the error goes away. TypeScript is correct here.

Comment (Gudahtt) [reply ui/components/multichain/pages/connections/connections.tsx:116]: In general, we should never use type assertions like this. Even in the scenarios where TypeScript lacks the necessary type information and there is no good way to provide it, there are better ways to "override" the type (type guards, the `satisfies` operator, type annotations, etc.`). The linked document has plenty more details. I'd encourage you to review it, and follow up with any questions you have about it.

But in this specific scenario, I don't think any override is needed

File: ui/pages/remove-snap-account/snap-account-card.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Gudahtt) [reply ui/pages/remove-snap-account/snap-account-card.tsx:28]: What would happen if the account was removed in another window while this was open?

[the remark this answers is not in the document]
Comment (Gudahtt) [reply ui/pages/remove-snap-account/snap-account-card.tsx:28]: There might still be a crash here, but probably it would have a minimal impact because it doesn't really make sense to remove an account that doesn't exist. I'll approve for now, but it might be a good idea to follow up here to improve how this missing account scenario is handled

**Review by Gudahtt — APPROVED** [review]
LGTM!
